worlds/clique/__init__.py

```diff
@@ -27,71 +24,70 @@ class CliqueWebWorld(WebWorld):
+
+        # Create locations.
+        for region_name, region_data in region_data_table.items():
+            region = self.multiworld.get_region(region_name, self.player)
+            region.add_locations({
+                location_name: location_data.address for location_name, location_data in location_data_table.items()
+                if location_data.region == region_name and location_data.can_create(self.multiworld, self.player)
+            }, CliqueLocation)
+            region.add_exits(region_data_table[region_name].connecting_regions)
+
+        # Place locked locations.
+        for location_name, location_data in locked_locations.items():
+            # Ignore locations we never created.
+            if not location_data.can_create(self.multiworld, self.player):
+                continue
+
+            locked_item = self.create_item(location_data_table[location_name].locked_item)
+            self.multiworld.get_location(location_name, self.player).place_locked_item(locked_item)
+
+        # Set priority location for the Big Red Button!
+        self.multiworld.priority_locations[self.player].value.add("The Big Red Button")
 
     def get_filler_item_name(self) -> str:
-        return self.multiworld.random.choice(self.item_name_to_id)
+        return "A Cool Filler Item (No Satisfaction Guaranteed)"
 
     def set_rules(self) -> None:
-        if self.multiworld.hard_mode[self.player]:
-            set_rule(
-                self.multiworld.get_location("The Big Red Button", self.player),
-                lambda state: state.has("Button Activation", self.player))
-
-        self.multiworld.completion_condition[self.player] = lambda state: \
-            state.has("Feeling of Satisfaction", self.player)
-
-
-def create_region(world: MultiWorld, player: int, name: str, locations=None, exits=None):
-    region = Region(name, player, world)
-    if locations:
-        for location_name in locations.keys():
-            region.locations.append(CliqueLocation(player, location_name, locations[location_name], region))
-
-    if exits:
-        for _exit in exits:
-            region.exits.append(Entrance(player, _exit, region))
-
-    return region
+        self.multiworld.get_location("The Big Red Button", self.player).access_rule =\
+            lambda state: has_unlocked_button(state, self.player)
+        self.multiworld.get_location("In the Player's Mind", self.player).access_rule =\
+            lambda state: has_unlocked_button(state, self.player)
+
+        # Do not allow button activations on buttons.
+        self.multiworld.get_location("The Big Red Button", self.player).item_rule =\
+            lambda item: item.name != "Button Activation"
+
+        # Completion condition.
+        self.multiworld.completion_condition[self.player] = lambda state: state.has("The Urge to Push", self.player)
+
+    def fill_slot_data(self):
+        return {
+            "color": getattr(self.multiworld, "color")[self.player].current_key
```

Comment:
Could make this a lot nicer and get rid of all the getattr with #993

Reply:
Yeah well... it ain't merged yet. \:P